Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerFragment.kt

```diff
@@ -350,70 +342,69 @@ class QrCodeScannerFragment : Fragment(R.layout.fragment_qrcode_scanner), AutoIn
-        MaterialAlertDialogBuilder(requireContext())
-            .setTitle(R.string.recycle_bin_restore_dgc_dialog_title)
-            .setCancelable(false)
-            .setMessage(R.string.recycle_bin_restore_dgc_dialog_message)
-            .setPositiveButton(android.R.string.ok) { _, _ -> viewModel.restoreCertificate(containerId) }
-            .show()
-    }
+    private fun showRestoreDgcConfirmation(containerId: CertificateContainerId) =
+        displayDialog(cancelable = false) {
+            setTitle(R.string.recycle_bin_restore_dgc_dialog_title)
+            setMessage(R.string.recycle_bin_restore_dgc_dialog_message)
+            setPositiveButton(android.R.string.ok) { _, _ -> viewModel.restoreCertificate(containerId) }
+        }
 
     private fun showRestoreCoronaTestConfirmation(recycledCoronaTest: BaseCoronaTest) =
-        RestoreCoronaTestConfirmationDialog
-            .showDialog(context = requireContext()) { viewModel.restoreCoronaTest(recycledCoronaTest) }
-
-    private fun showMaxPersonExceedsThresholdResult(max: Int, deeplink: Uri, navOptions: NavOptions) {
-        MaterialAlertDialogBuilder(requireContext())
-            .setTitle(R.string.qr_code_error_max_person_threshold_title)
-            .setCancelable(false)
-            .setMessage(getString(R.string.qr_code_error_max_person_threshold_body, max))
-            .setOnDismissListener {
-                findNavController().navigate(deeplink, navOptions)
+        displayDialog(cancelable = false) {
+            setTitle(R.string.recycle_bin_restore_corona_test_dialog_title)
+            setMessage(R.string.recycle_bin_restore_corona_test_dialog_message)
+            setPositiveButton(R.string.recycle_bin_restore_corona_test_dialog_title) { _, _ ->
+                viewModel.restoreCoronaTest(
+                    recycledCoronaTest
+                )
             }
-            .setPositiveButton(R.string.qr_code_error_max_person_covpasscheck_button) { _, _ ->
+        }
+
+    private fun showMaxPersonExceedsThresholdResult(max: Int, deeplink: Uri, navOptions: NavOptions) =
+        displayDialog(cancelable = false) {
+            setTitle(R.string.qr_code_error_max_person_threshold_title)
+            setMessage(getString(R.string.qr_code_error_max_person_threshold_body, max))
+            setPositiveButton(R.string.qr_code_error_max_person_covpasscheck_button) { _, _ ->
                 openUrl(R.string.qr_code_error_max_person_covpasscheck_link)
             }
-            .setNegativeButton(R.string.qr_code_error_max_person_faq_button) { _, _ ->
+            setNegativeButton(R.string.qr_code_error_max_person_faq_button) { _, _ ->
                 openUrl(R.string.qr_code_error_max_person_faq_link)
             }
-            .setNeutralButton(android.R.string.ok) { _, _ -> }
-            .show()
-    }
+            setNeutralButton(android.R.string.ok) { _, _ ->
+                findNavController().navigate(deeplink, navOptions)
+            }
+        }
```

Comment:
`setOnDismissListener` is missing after refactoring it

Reply:
Fixed it.